mypy/semanal.py

```diff
@@ -644,6 +645,19 @@ def add_implicit_module_attrs(self, file_node: MypyFile) -> None:
                     self.defer()
                     return
                 typ = inst
+            elif name == "__spec__":
+                inst: Type | None = self.named_type_or_none("importlib.machinery.ModuleSpec")
+                if inst is None:
+                    if self.final_iteration:
```

Comment:
I wonder if this may down slow down tests significantly, since every module top level will have to be processed many times? Maybe instead check `options.use_builtins_fixtures` and fall back to `builtins.object` immediately if it's true.